app/src/test/java/org/simple/clinic/summary/PatientSummaryScreenControllerTest.kt

```diff
@@ -346,74 +346,92 @@ class PatientSummaryScreenControllerTest {
-    } else {
-      if (patientSummaryCaller == PatientSummaryCaller.NEW_PATIENT) {
-        verify(screen).goBackToHome()
-      } else {
-        verify(screen).goBackToPatientSearch()
-      }
-    }
+    verify(screen, never()).goBackToPatientSearch()
+    verify(screen, never()).goBackToHome()
+    verify(screen).showScheduleAppointmentSheet(patientUuid)
   }
 
   @Test
-  @Parameters(method = "bpSavedAndPatientSummaryCallers")
-  fun `when save button is clicked, then user should be taken back to the home screen, or schedule appointment sheet should open`(
-      wasBloodPressureSaved: Boolean,
+  @Parameters(method = "patient summary callers")
+  fun `when there are patient summary changes and all bps are deleted, clicking on back must go back`(
       patientSummaryCaller: PatientSummaryCaller
   ) {
-    uiEvents.onNext(PatientSummaryScreenCreated(patientUuid, caller = patientSummaryCaller, screenCreatedTimestamp = Instant.now(clock)))
-    uiEvents.onNext(PatientSummaryBloodPressureClosed(wasBloodPressureSaved))
-    uiEvents.onNext(PatientSummaryDoneClicked())
+    val patientSummaryItems = mock<PatientSummaryItems>()
+    whenever(patientSummaryItems.hasItemChangedSince(any())).thenReturn(true)
 
-    if (wasBloodPressureSaved) {
-      verify(screen).showScheduleAppointmentSheet(patientUuid)
-      verify(screen, never()).goBackToHome()
-    } else {
+    uiEvents.onNext(PatientSummaryScreenCreated(
+        patientUuid = patientUuid,
+        caller = patientSummaryCaller,
+        screenCreatedTimestamp = Instant.now(clock)
+    ))
+    uiEvents.onNext(PatientSummaryItemChanged(patientSummaryItems))
+    uiEvents.onNext(PatientSummaryAllBloodPressuresDeleted(true))
+    uiEvents.onNext(PatientSummaryBackClicked())
+
+    verify(screen, never()).showScheduleAppointmentSheet(patientUuid)
+    if (patientSummaryCaller == PatientSummaryCaller.NEW_PATIENT) {
       verify(screen).goBackToHome()
-      verify(screen, never()).showScheduleAppointmentSheet(any())
+    } else {
+      verify(screen).goBackToPatientSearch()
     }
   }
 
   @Test
-  @Parameters(method = "bpSavedAndPatientSummaryCallers")
-  fun `when summary screen is restored, and bp was saved earlier, schedule appointment sheet should open, on clicking back or done`(
-      wasBloodPressureSaved: Boolean,
+  @Parameters(method = "patient summary callers")
+  fun `when there are no patient summary changes and all bps are not deleted, clicking on back must go back`(
       patientSummaryCaller: PatientSummaryCaller
   ) {
-    uiEvents.onNext(PatientSummaryScreenCreated(patientUuid, caller = patientSummaryCaller, screenCreatedTimestamp = Instant.now(clock)))
-    uiEvents.onNext(PatientSummaryRestoredWithBPSaved(wasBloodPressureSaved))
-    uiEvents.onNext(PatientSummaryDoneClicked())
+    val patientSummaryItems = mock<PatientSummaryItems>()
+    whenever(patientSummaryItems.hasItemChangedSince(any())).thenReturn(false)
```

Comment:
nitpick: you've grouped these two lines together in some functions and kept a line break in between in some functions. Please remove this break if you get a chance to touch this class again?

Reply:
There's no line break in the code. Is this related to GitHub's diff viewer?

Reply:
Yes, my bad sorry